### PR TITLE
Added Rspec and few other gems to testing evironment

### PR DESCRIPTION
**Pull request by neosb**
1. I have added this gems to `:test`:
   - rspec-rails
   - factory_girl_rails
   - shoulda-matchers
   - database_cleaner
2. Then I have invoked `bundle`, `rails generate rspec:install` and `bundle binstubs rspec-core`

@cintrzyk I would left autogenerated comments where they are, because we are just starting with testing. I have looked throgh them and they let us configure Rspec. I don't if we will need them?

### CODE REVIEW

File: spec/rails_helper.rb [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+  # You can disable this behaviour by removing the line below, and instead
+  # explicitly tag your specs with their type, e.g.:
+  #
+  #     RSpec.describe UsersController, :type => :controller do
+  #       # ...
+  #     end
+  #
+  # The different available types are documented in the features, such as in
+  # https://relishapp.com/rspec/rspec-rails/docs
+  config.infer_spec_type_from_file_location!
```
Comment (cintrzyk) [inline spec/rails_helper.rb:19]: please remove all the comments and add database_cleaner config :)

File: spec/spec_helper.rb [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+  # Print the 10 slowest examples and example groups at the
+  # end of the spec run, to help surface which specs are running
+  # particularly slow.
+  config.profile_examples = 10
+
+  # Run specs in random order to surface order dependencies. If you find an
+  # order dependency and want to debug it, you can fix the order by providing
+  # the seed, which is printed after each run.
+  #     --seed 1234
+  config.order = :random
```
Comment (cintrzyk) [inline spec/spec_helper.rb:10]: random specs are nice :D please enable this option

File: bin/rspec [hunk 3]
```diff
@@ -0,0 +1,16 @@
...
+#
+
+require 'pathname'
+ENV['BUNDLE_GEMFILE'] ||= File.expand_path("../../Gemfile",
+  Pathname.new(__FILE__).realpath)
+
+require 'rubygems'
+require 'bundler/setup'
+
+load Gem.bin_path('rspec-core', 'rspec')
```
Comment (cintrzyk) [inline bin/rspec:7]: this is a standard rspec binary file. Lets add Spring to it ;)

File: spec/rails_helper.rb (hunk 1 above)
Comment (neosb) [inline spec/rails_helper.rb:19]: I will use [this tutorial](http://devblog.avdi.org/2012/08/31/configuring-database_cleaner-with-rails-rspec-capybara-and-selenium/) to setup database_cleaner

File: bin/rspec (hunk 3 above)
Comment (cintrzyk) [inline bin/rspec:7]: please take a look at gem [spring-commands-rspec](https://github.com/jonleighton/spring-commands-rspec)

Comment (neosb) [inline bin/rspec:7]: I have added it to Gemfile, in 9587b7d

File: spec/spec_helper.rb (hunk 2 above)
Comment (neosb) [inline spec/spec_helper.rb:10]: I removed comments from `spec/spec_helper` too

**Comment by neosb** [comment]
In 9c6da90 I have added test for routing root of app to the `home#index`. Looks like tests are all set.

File: bin/rspec (hunk 3 above)
Comment (cintrzyk) [inline bin/rspec:7]: I think part of the code can be removed:

``` ruby
require 'pathname'
ENV['BUNDLE_GEMFILE'] ||= File.expand_path("../../Gemfile",
  Pathname.new(__FILE__).realpath)

require 'rubygems'
require 'bundler/setup'
```

File: spec/support/database_cleaner.rb [hunk 4]
```diff
@@ -0,0 +1,26 @@
...
+    # Start transaction
+    DatabaseCleaner.start
+
+    # Run example
+    example.run
+
+    # Rollback transaction
+    DatabaseCleaner.clean
+  end
+end
```
Comment (cintrzyk) [inline spec/support/database_cleaner.rb]: please move this config into `rails_helper.rb`. We can place some other config not only database_cleaner inside `before` `around` blocks in the future :)

File: spec/rails_helper.rb [hunk 5]
```diff
@@ -0,0 +1,34 @@
...
+  config.use_transactional_fixtures = false
+
+
+  config.infer_spec_type_from_file_location!
+
+  config.before(:suite) do
+    DatabaseCleaner.clean_with(:truncation)
+    
+    # uncomment next line to seed databse before tests
+    # load "#{Rails.root}/db/seeds.rb"
```
Comment (cintrzyk) [inline spec/rails_helper.rb]: we don't need it

File: spec/spec_helper.rb [hunk 6]
```diff
@@ -0,0 +1,14 @@
+require 'codeclimate-test-reporter'
+CodeClimate::TestReporter.start if ENV['CI']
```
Comment (cintrzyk) [inline spec/spec_helper.rb]: `CodeClimate` should be configured in `rails_helper`

File: spec/rails_helper.rb (hunk 5 above)
Comment (neosb) [inline spec/rails_helper.rb]: you mean the comment about seeds.rb?

Comment (cintrzyk) [inline spec/rails_helper.rb]: yes, please remove it :) We will use factories to create records

**Comment by neosb** [comment]
I have updated `feature/rspec-setup` branch from `master` branch, that's why all this commits are here.

**Comment by neosb** [comment]
Fixed doubled commits.

**Comment by cintrzyk** [comment]
:cake:

**Pull request merged**